src/colmap/scene/database.cc

```diff
@@ -451,8 +451,10 @@ PosePrior Database::ReadPosePrior(const image_t image_id) const {
   if (rc == SQLITE_ROW) {
     prior.position =
         ReadStaticMatrixBlob<Eigen::Vector3d>(sql_stmt_read_pose_prior_, rc, 1);
+    prior.position_covariance =
+        ReadStaticMatrixBlob<Eigen::Matrix3d>(sql_stmt_read_pose_prior_, rc, 2);
     prior.coordinate_system = static_cast<PosePrior::CoordinateSystem>(
-        sqlite3_column_int64(sql_stmt_read_pose_prior_, 2));
+        sqlite3_column_int64(sql_stmt_read_pose_prior_, 3));
```

Comment:
Unfortunately it is not so reading a database with a different data structure for its pose_priors table will lead to a SQL logic error.  Any idea on how to make it backward compatible?  Otherwise we could probably update the database migration python scripts.

Reply:
This makes sense, thank you!